### ISSUE TITLE
Error text when creating Instructor task from recruitment

### ISSUE THREAD

**Issue by maneesha** [carpentries/amy#2449]
As identified in this [private slack conversation](https://swcarpentry.slack.com/archives/GGE8UA04X/p1687869107693479):

When an admin tries to accept an instructor signup to create an instructor task, if that instructor already has an Instructor task for that event, we get this not very useful error:
![image](https://github.com/carpentries/amy/assets/829690/97637b83-7985-40a2-bf8b-8ae986df2948)

We should  consider how we handle a signup if the Person already has an Instructor task for that event. Suggestion: Clicking "accept" gives a notification that the Person already has an instructor task, and maps that acceptance to that task.  Clicking "decline" gives an error, requiring the admin to remove the associated Instructor task before continuing.  

Tagging @danielle06  and @elichad  for feedback.

**Comment by elichad** [comment]
Technical notes:

This error arises when an IntegrityError occurs while handling a signup: https://github.com/carpentries/amy/blob/39f9544c8edfc50a3abc9a3fc5d07f77d7d71cfd/amy/recruitment/views.py#L375-L396

I _think_ we can expect most (if not all) of these IntegrityErrors to come from a violation of our constraint that any Task should have a unique combination of event, person, and role (i.e. instructor already has a task for this event). 
The other way would be for the Task to be provided with a non-existent event/person/role key.

In any case, we should be able to perform this check for an existing task as part of this try/catch loop for the "accept" part.
For "decline," we already quietly remove the associated Instructor task if present.

**Comment by maneesha** [comment]
On **_decline_**, we should not quietly remove the associated Instructor task if present -- there should be an explicit warning that this is happening and the recruitment state should change to decline.

On **_accept_**, we should have a notification that the task already exists and the recruitment state should change to accepted.  (the second part looks like it is already happening)

**Comment by danielle06** [comment]
Hi @maneesha and @elichad , thank you so much for prioritizing this. The changes to the process, outlined in Maneesha's last comment, make the workflow very clear and uncomplicated.

**Comment by pbanaszkiewicz** [comment]
@maneesha If I change the behavior for "decline", then there will be a problem with "instructor confirmed for workshop" email.

This email action currently would be cancelled if it had been scheduled, the task had existed, and user clicked "decline".

After the proposed change for "decline", the possible email action would remain and admin would have to manually go and remove task or cancel the scheduled email.


In my opinion, we should split "instructor confirmed for workshop" email action into two separate actions:

1. email action (action A) confirming that instructor task was created for a workshop
2. email action (action B) confirming that instructor was accepted in specific recruitment.

Those actions don't need to overlap, so even though accepting instructor application creates a task for the instructor, we can select which email action is run then (so in this case only action B).

Once the "instructor confirmed for workshop" email action is split, the problem outlined above ("the possible email action would remain and admin would have to manually go and remove task or cancel the scheduled email") disappears, because only action B is related to handling signups, and in this case signup was declined, so action B should be cancelled.


This problem with "instructor confirmed for workshop" email action is something I have been thinking for a while now and in my opinion splitting this action should lead to cleaner logic and cleaner email action handling.

**Comment by maneesha** [comment]
I agree with splitting this.
